### PR TITLE
Add basic responsive functionality

### PR DESCRIPTION
**Pull request by anthonydillon**
## Done
Added basic responsive functionality by styling the global nav to the bottom of the page on small screens.

## QA
- Pull down code
- Install by running `npm i` 
- Run `gulp test` and see there are no errors
- Then build using `gulp build`
- Open the index.html page in your browser
- Shrink the screen down to < 768px and see that the global nav jumps to the bottom and becomes an accordion. Like on http://www.ubuntu.com

## Details
Fixes https://github.com/canonical-webteam/global-nav/issues/14 https://github.com/canonical-webteam/global-nav/issues/2 https://github.com/canonical-webteam/global-nav/issues/1

### CODE REVIEW

**Comment by richmccartney** [comment]
@anthonydillon There is a warning with Scss lint that should be addressed

```bash
src/sass/_patterns-global-nav.scss
  132:17  warning  @extend must be used with a %placeholder  placeholder-in-extend

✖ 1 problem (0 errors, 1 warning)
```

File: src/sass/_patterns-global-nav.scss [hunk 1]
```diff
@@ -1,74 +1,155 @@
...
       }
+    }
 
-      @media only screen and (min-width: 984px) {
-        width: 984px !important;
+    &__title {
+      display: none;
+
+      @media only screen and (max-width: 768px) {
+        background-image: url('#{$global-asset-path}7bd1bd7b-arrow_down_9fa097.svg');
```
Comment (richmccartney) [inline src/sass/_patterns-global-nav.scss]: Asset isn't loading on Chrome.

File: src/sass/_patterns-global-nav.scss [hunk 2]
```diff
@@ -1,74 +1,155 @@
...
+        display: block;
+        font-size: .8125rem;
+        font-weight: 300;
+        line-height: 1.5;
+        margin-bottom: 0;
+        padding: .625rem 0 .625rem 1.875rem;
+        position: relative;
+
+        &.is-revealed {
+          background-image: url('#{$global-asset-path}c65b8106-icon-arrow-up.svg');
```
Comment (richmccartney) [inline src/sass/_patterns-global-nav.scss]: Asset isn't loading on Chrome.

File: src/sass/_patterns-global-nav.scss [hunk 3]
```diff
@@ -1,74 +1,155 @@
...
+        }
+
+        &::before {
+          background-image: url('#{$global-asset-path}e1b12fdd-external-link-9fa097.svg');
```
Comment (richmccartney) [inline src/sass/_patterns-global-nav.scss]: Asset isn't loading on Chrome.

**Comment by anthonydillon** [comment]
Yea I'm aware of that one. Its a complicated one as it touches a lot of the
codebase. I will catch that in a follow up a its just a warning.

File: src/sass/_settings-colors.scss [hunk 4]
```diff
@@ -1,5 +1,9 @@
 $global-white:        #fff;
+$global-black:        #000;
 $global-background:   #efefef;
 $global-grey:         #d7d7d7;
+$global-mid-grey:     #cdcdcd;
 $global-dark-grey:    #333;
+$global-x-dark-grey:  #666;
 $global-orange:       #e95420;
+$global-transparent:  transparent;
```
Comment (richmccartney) [inline src/sass/_settings-colors.scss]: If this is a module we would be using on different projects should we add ```!default``` to these variables? Allows easier overrides if it's included in other projects.

File: src/js/global.js [hunk 5]
```diff
@@ -156,6 +157,20 @@ ubuntu.globalNav = function() {
...
+          'click',
+          function(smallScreenToggle) {
+            return function(clickEvent) {
+              navList.classList.toggle('is-revealed');
+              smallScreenToggle.classList.toggle('is-revealed');
+            };
+          }(smallScreenToggle)
+        );
+      }
     },
```
Comment (richmccartney) [inline src/js/global.js]: With this code if you were on a desktop browser and toggle the list, it'll stay open from small screen to large and can not resize. Should we think about checking if the screen is being resized and remove any toggles?

screen shot 2017-05-02 at 16 08 10

File: src/js/global.js [hunk 6]
```diff
@@ -77,6 +77,7 @@ ubuntu.globalNav = function() {
       let wrapper = createFromHTML(
         '<nav class="global-nav">' +
         '  <div class="global-nav__wrapper">' +
+        '    <h2 class="global-nav__title">Ubuntu websites</h2>' +
```
Comment (richmccartney) [inline src/js/global.js]: This should be a ```<button>``` element with aria tags stating what it controls. Anything which is interactive but not a link is advised to be a button element.

File: src/sass/_patterns-global-nav.scss [hunk 7]
```diff
@@ -1,74 +1,155 @@
...
+    top: 0;
+    width: 100%;
+    z-index: 101;
+
+    @media only screen and (max-width: 768px) {
+      background: $global-transparent;
+      bottom: 0;
+      box-shadow: none;
+      top: auto;
+    }
```
Comment (richmccartney) [inline src/sass/_patterns-global-nav.scss]: I think the overall positioning needs a rethink, when adding a lot of content it'll sit in the middle of any content there. I feel we should be using a different approach here. 

Here is a screenshot showing the position when there are 20 paragraph elements on the page.
screen shot 2017-05-02 at 16 26 50

**Comment by anthonydillon** [comment]
@richmccartney thanks for the detailed review! All thought and fixed I believe. Ready for another review, please.

File: src/js/global.js [hunk 8]
```diff
@@ -156,6 +159,22 @@ ubuntu.globalNav = function() {
...
+      let navList = globalNav.querySelector('.global-nav__list');
+      if (smallScreenToggle && navList) {
+        smallScreenToggle.addEventListener(
+          'click',
+          function(smallScreenToggle) {
+            return function(clickEvent) {
+              smallScreenToggle.classList.toggle('is-revealed');
+              smallScreenToggle.setAttribute('aria-expanded', true);
+              navList.classList.toggle('is-revealed');
+              navList.setAttribute('aria-hidden', false);
```
Comment (richmccartney) [inline src/js/global.js]: Once clicked again this attribute doesn't toggle back to true. I think there might need to be a little bit of logic needed just to make sure the navList changes from false to true.

File: src/sass/_patterns-global-nav.scss [hunk 9]
```diff
@@ -1,171 +1,307 @@
...
+      display: none;
+
+      @media only screen and (max-width: $global-breakpoint-medium) {
+        background-color: transparent;
+        background-image: url('#{$global-asset-path}7bd1bd7b-arrow_down_9fa097.svg');
+        background-position: calc(100% - .625rem) 50%;
+        background-repeat: no-repeat;
+        background-size: .8125rem;
+        border-bottom: 1px solid $global-mid-grey;
+        border-right: 0;
```
Comment (richmccartney) [inline src/sass/_patterns-global-nav.scss]: Need to add ```border-left: 0``` as I can see it on small screen 
screen shot 2017-05-03 at 10 28 06

File: src/sass/_patterns-global-nav.scss [hunk 10]
```diff
@@ -1,171 +1,307 @@
...
+      height: $navigaiton-height;
       margin: 0;
-      height: 30px;
       margin-top: -1px;
       position: relative;
+      text-align: left;
       top: 0;
 
+      @media only screen and (max-width: $global-breakpoint-medium) {
+        border-bottom: 1px solid $global-mid-grey;
```
Comment (richmccartney) [inline src/sass/_patterns-global-nav.scss]: If its the last item and there is only one, it gives a little border which feels a little odd. Should we change it to be ```border-top``` instead? 
screen shot 2017-05-03 at 10 28 15

File: src/js/global.js (hunk 8 above)
Comment (anthonydillon) [inline src/js/global.js]: Good spot, fixed.

File: src/sass/_patterns-global-nav.scss (hunk 9 above)
Comment (anthonydillon) [inline src/sass/_patterns-global-nav.scss]: Ah I dont see it but its there ... removed it.

File: src/sass/_patterns-global-nav.scss (hunk 10 above)
Comment (anthonydillon) [inline src/sass/_patterns-global-nav.scss]: Bottom border is gone.

**Comment by anthonydillon** [comment]
@richmccartney thanks for the spots, the borders should now work for all cases and order if lists. Ready for review again.

**Comment by richmccartney** [comment]
+1 perfect looks awesome now dude!

**Pull request merged**